util/net/http_transport_mac.mm

```diff
@@ -289,10 +268,122 @@ - (BOOL)setProperty:(id)property forKey:(NSStreamPropertyKey)key {
   }
 }
 
+bool HTTPTransportMac::ExecuteProxyRequest(NSMutableURLRequest* request,
+                                           const std::string& proxy,
+                                           std::string* response_body) {
+  __block bool sync_rv = false;
+  @autoreleasepool {
+    NSURLSessionConfiguration* sessionConfig =
+        [NSURLSessionConfiguration ephemeralSessionConfiguration];
+
+    std::string scheme, host, port, rest_ignored;
+    CrackURL(proxy, &scheme, &host, &port, &rest_ignored);
+    NSString* schemeNS = base::SysUTF8ToNSString(scheme);
+    NSString* hostNS = base::SysUTF8ToNSString(host);
+    NSNumber* proxy_port = @(std::stoi(port));
```

Comment:
If parsing the proxy string fails, the SDK would crash mid-flight right? I'm not saying that we should cover every possible error case, as people configuring a proxy should generally know what they are doing but I'm just wondering 😅

Reply:
True, if parsing fails, the `crashpad_handler` will stop and log an error (if it was a debug build). There is not a lot that we can do at that point. If we want to make that developer friendlier by checking the URL in any way, we should do it in `sentry_options_set_http_proxy()` and not in the `crashpad_handler`, the expectation here should be that the URL was correct when it was passed to the handler.